source/io/PropertyNode.cpp

```diff
@@ -15,18 +15,22 @@ SDG
 #include "ExecutionContext.h"
 #include "StringUtilities.h"
 #include "TDCodecVia.h"
+#include "VirtualInstrument.h"
 #include <stdio.h>
 
+
+#define VIREO_TYPE_PropertyNode 1
+
 #if defined (VIREO_TYPE_PropertyNode)
 namespace Vireo {
 
 #if kVireoOS_emscripten
 extern "C" {
     // JavaScript function prototypes
-    // Parameters: viName, controlId, propertyName, value, errorStatus*, errorCode*, errorSource*
-    extern void jsPropertyNodeWrite(StringRef, StringRef, StringRef, void *, Boolean *, Int32 *, StringRef);
-    // Parameters: viName, controlId, propertyName, value, errorStatus*, errorCode*, errorSource*
-    extern void jsPropertyNodeRead(StringRef, StringRef, StringRef, void *, Boolean *, Int32 *, StringRef);
+    // Parameters: viName, controlId, propertyName, propertyType, propertyPath, errorStatus*, errorCode*, errorSource*
```

Comment:
I thought controlId would be resolved at JS layer from the path (viName and dataItem name). Is that changing?

Reply:
shouldn't be changing, dataItemId might be a better name

Reply:
I guess I should change the name to be dataItemName instead.

Reply:
As it is now, without control references, we pass the viName and controlId (this could be the dataItemName). Once we have control references we would pass a control reference where this information would be encoded. i.e. we are expeting this to change.

Reply:
dataItemId will be